search/indexing_api.py

```diff
@@ -370,34 +369,60 @@ def recreate_index():
     """
     Wipe and recreate index and mapping, and index all items.
     """
-    temp_index = get_temp_index()
-    clear_index(temp_index)
+    conn = get_conn(verify=False)
+
+    # Create new backing index for reindex
+    new_backing_index = "{index}_{uuid}".format(
+        index=settings.ELASTICSEARCH_INDEX,
+        uuid=uuid4().hex,
+    )
+    clear_index(new_backing_index)
+
+    # Clear away temp alias so we can reuse it
+    temp_alias = get_temp_alias()
+    if conn.indices.exists_alias(name=temp_alias):
+        # Deletes both alias and backing indexes
+        conn.indices.delete(temp_alias)
+
+    # Point temp_alias toward new backing index
+    conn.indices.put_alias(index=new_backing_index, name=temp_alias)
+
+    # Do the indexing on the temp index
     start = datetime.now(pytz.UTC)
     log.info("Indexing %d program enrollments...", ProgramEnrollment.objects.count())
-    index_program_enrolled_users(ProgramEnrollment.objects.iterator(), [temp_index])
+    index_program_enrolled_users(ProgramEnrollment.objects.iterator(), [new_backing_index])
```

Comment:
I would suggest wrapping the indexing calls in a `try..finally` and clean up the temp index/alias in the finally block.